sw/airborne/firmwares/rotorcraft/autopilot.c

```diff
@@ -573,6 +572,77 @@ bool autopilot_guided_move_ned(float vx, float vy, float vz, float heading)
   return false;
 }
 
+/* Note: Offset position command in NED frame or body frame will only be implemented if
+ * local reference frame has been initialised. Additionally, if both the offset NED
+ * and offset body flags are set, the x,y setting is not implemented.
+ * Flag definition:
+   bit 0: x,y as offset coordinates
+   bit 1: x,y in body coordinates
+   bit 2: z as offset coordinates
+   bit 3: yaw as offset coordinates
+   bit 4: free
+   bit 5: x,y as vel
+   bit 6: z as vel
+   bit 7: yaw as rate
+ */
+void autopilot_guided_update(uint8_t flags, float x, float y, float z, float yaw)
+{
+  /* only update setpoints when in guided mode */
+  if (autopilot_mode != AP_MODE_GUIDED) {
+    return;
+  }
+
+  // handle x,y
+  if (bit_is_set(flags, 5)) { // velocity setpoint
+    if (bit_is_set(flags, 1)) { // set velocity in body frame
+      float psi = stateGetNedToBodyEulers_f()->psi;
+      x =  cosf(-psi) * x + sinf(-psi) * y;
+      y = -sinf(-psi) * x + cosf(-psi) * y;
+    }
+    guidance_h_set_guided_vel(x, y);
+  } else {  // position setpoint
+    if ((flags & 3) == 0) {   // set absolute position setpoint
+      guidance_h_set_guided_pos(x, y);
```

Comment:
yes and no. Here, the offset bit flag is used to indicate it is offset in NED frame and the body flag is used to say it is offset in body frame. I think this is obvious as you can't have absolute in body frame. Are the comments I wrote not sufficient? 

I can alternatively change it that if the body flag is set ignore the other flags and just set the offset body. If the body flag isn't set then I check if it should be offset or absolute in NED... Would that be more reasonable?